stdlib/public/Darwin/Foundation/String.swift

```diff
@@ -26,6 +26,25 @@ extension String {
   }
 }
 
+@_effects(readonly)
```

Comment:
TODO for tomorrow: check if these three readonly wrappers are actually necessary